### PR TITLE
dpgen

### PR DESCRIPTION
**Pull request by njzjz**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by njzjz** [comment]
Please review, thank you! @conda-forge/help-python

File: recipes/dpgen/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+requirements:
+  host:
+    - python >=3.5
+    - pip
+    - m2r
+  run:
+    - python >=3.5
+    - numpy >=1.14.3
+    - dpdata>=0.1.3
+    - pymatgen>=2017.9.1
```
Comment (beckermr) [inline recipes/dpgen/meta.yaml]: add spaces between the package name and the `>=`

File: recipes/dpgen/add_version.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/setup.py b/setup.py
```
Comment (beckermr) [inline recipes/dpgen/add_version.patch]: What is the purpose of the version patch here?

Comment (njzjz) [reply recipes/dpgen/add_version.patch]: The origin program uses `setuptools-git-version` to read the version from the git tag, but there is no git tags in a `.tar.gz` file. I have to add the version into `setup.py`.

**Pull request merged**